src/core/tsi/alts/handshaker/alts_handshaker_client.cc

```diff
@@ -595,6 +605,14 @@ static grpc_byte_buffer* get_serialized_start_server(
       server_version, arena.ptr(), &client->options->rpc_versions);
   grpc_gcp_StartServerHandshakeReq_set_max_frame_size(
       start_server, static_cast<uint32_t>(client->max_frame_size));
+  grpc_gcp_TransportProtocolPreferences* preferences =
```

Comment:
We should only allocate this if `client->preferred_transport_protocols` is populated (to avoid having to distinguish between a proto that is "not set" vs. "empty").